packages/Liquid-Tests.package/LQParticipantMenuTests.class/instance/tearDown.st

```diff
@@ -0,0 +1,5 @@
+running
+tearDown
+
+	super tearDown.
+	LiquidParticipantMenu answeredPollIds: Set new. 
```

Comment:
[boulder] I think we should reset in either setUp or Teardown, but not both - double-resetting might hide valid bugs.

Reply:
As far as I can see @PaulVII wrote this test. After thinking about it for a short while, I think the reset in setUp is done to obviously clear the Set for the test, but I believe they reset it again in tearDown to make sure it doesn't interfere with the "normal" Liquid, since the Poll-ID's used in the test would be stored indefinitely, or until running setUp again. Correct me if I'm wrong

Reply:
if every test that depends on `LiquidParticipantMenu answeredPollIds` reset it during `setUp`, that should *technically* be enough. if we also reset during `teardown`, that could result in a future test forgetting to clean their dependencies but still passing. resetting twice is defensive programming

Reply:
I agree. Especially since we assume that our Users don't touch code, meaning they will not run the tests, we should not have any problems with only resetting in setUp. I however don't see what you mean with 

> that could result in a future test forgetting to clean their dependencies but still passing

How would leaving out tearDown result in such a test failing when forgetting to clean up behind it?

Reply:
Yes, we did that, I also added a comment on some other file explaining why and asking for feedback^^

Reply:
Assume a test cases A that requires some shared global state to be blank for its tests to pass. It implements `setUp` and `teardown` to reset global state.

Now another developer writes a new test case B, it too depends on the same global state - but they're not aware of it. Since `A >> tearDown` cleared the state for them, everything's fine - as long as A runs before B. Once the execution order changes, or B is run two times in a row, it fails.

If A didn't implement `tearDown`, then the developer of `B` would've been more likely to stumbled over the shared global state, and would've remembered to implement `setUp`  themselves.

Thus, counterintuitively, implementing `tearDown` helps hiding faulty test code instead of making it obvious.

Reply:
Top, thanks for the example. I was still mentally "locked up" in the current class, where we clear the global state in setUp anyways. Still, maybe @PaulVII wants to add his opinion before we change this.

Reply:
As I understand it, @Skn0tt advocates for just using setUp and never teardown right? I'd be fine with that, it solves my problem too^^

Reply:
Done, I have deleted the tearDown-Method